hydrolib/core/dflowfm/extold/models.py

```diff
@@ -0,0 +1,417 @@
+    FrictionCoefficient = "frictioncoefficient"
+    """Friction coefficient"""
+    HorizontalEddyViscosityCoefficient = "horizontaleddyviscositycoefficient"
+    """Horizontal eddy viscosity coefficient"""
+    InternalTidesFrictionCoefficient = "internaltidesfrictioncoefficient"
+    """Internal tides friction coefficient"""
+    HorizontalEddyDiffusivityCoefficient = "horizontaleddydiffusivitycoefficient"
+    """Horizontal eddy diffusivity coefficient"""
+    AdvectionType = "advectiontype"
+    """Type of advection scheme"""
+    IBotLevType = "ibotlevtype"
+    """Type of bed-level handling"""
+
+    # Miscellaneous
+    ShiptXY = "shiptxy"
+    """shiptxy"""
+    MovingStationXY = "movingstationxy"
+    """Moving observation point for output (time, x, y)"""
+    WaveSignificantHeight = "wavesignificantheight"
+    """Wave significant heigth"""
+    WavePeriod = "waveperiod"
+    """Wave period"""
+
+
+class FileType(IntEnum):
+    """Enum class containing the valid values for the `filetype` attribute
+    in the [ExtForcing][hydrolib.core.dflowfm.extold.models.ExtForcing] class.
+    """
+
+    TimeSeries = 1
+    """1. Time series"""
+    TimeSeriesMagnitudeAndDirection = 2
+    """2. Time series magnitude and direction"""
+    SpatiallyVaryingWeather = 3
+    """3. Spatially varying weather"""
+    ArcInfo = 4
+    """4. ArcInfo"""
+    SpiderWebData = 5
+    """5. Spiderweb data (cyclones)"""
+    CurvilinearData = 6
+    """6. Curvilinear data"""
+    Samples = 7
+    """7. Samples"""
+    TriangulationMagnitudeAndDirection = 8
+    """8. Triangulation magnitude and direction"""
+    Polyline = 9
+    """9. Polyline (<*.pli>-file)"""
+    NetCDFGridData = 11
+    """11. NetCDF grid data (e.g. meteo fields)"""
+    NetCDFWaveData = 14
+    """14. NetCDF wave data"""
+
+
+class Method(IntEnum):
+    """Enum class containing the valid values for the `method` attribute
+    in the [ExtForcing][hydrolib.core.dflowfm.extold.models.ExtForcing] class.
+    """
+
+    PassThrough = 1
+    """1. Pass through (no interpolation)"""
+    InterpolateTimeAndSpace = 2
+    """2. Interpolate time and space"""
+    InterpolateTimeAndSpaceSaveWeights = 3
+    """3. Interpolate time and space, save weights"""
+    InterpolateSpace = 4
+    """4. Interpolate space"""
+    InterpolateTime = 5
+    """5. Interpolate time"""
+    Averaging = 6
+    """6. Averaging"""
+    InterpolateExtrapolateTime = 7
+    """7. Interpolate/Extrapolate time"""
+
+
+class Operand(str, Enum):
+    """Enum class containing the valid values for the `operand` attribute
+    in the [ExtForcing][hydrolib.core.dflowfm.extold.models.ExtForcing] class.
+    """
+
+    OverwriteExistingValues = "O"
+    """Existing values are overwritten."""
+    SuperimposeNewValues = "+"
+    """New values are superimposed."""
+
+
+class ExtForcing(BaseModel):
+    """Class holding the external forcing values."""
+
+    quantity: Union[Quantity, str] = Field(alias="QUANTITY")
```

Comment:
Why is this a `Union[Quantity, str]`? Only to support user-defined tracers I guess?

Reply:
I do not! :(